backbone.js

```diff
@@ -1994,9 +1994,31 @@
       if (this.fragment === decodedFragment) return;
       this.fragment = decodedFragment;
 
+      featuredItemOverride: {
+        if (!state || !state.moduleFetchParams) {
+          break featuredItemOverride;
+        }
+
+        var featuredItem = state.moduleFetchParams.featured_item;
+
+        if (!featuredItem || !featuredItem.id) {
+          break featuredItemOverride;
+        }
+
+        // prevent `null`, which causes puma to crash
+        delete state.moduleFetchParams.featured_item;
+
+        var featuredItemKeys = Object.keys(featuredItem);
+        for (var i = 0, key, newKey; i < featuredItemKeys.length; i++) {
+          key = featuredItemKeys[i];
+          newKey = 'featured_item[' + featuredItem.id + '][' + key + ']';
+          state.moduleFetchParams[newKey] = featuredItem[key];
```

Comment:
this modifies the original state object that was passed in, this may not be safe. I don't have enough context to say in this case, but wanted to call it out